lib/version/version.go

```diff
@@ -0,0 +1,52 @@
+// Package version provides machinery for versioning file names
+// with a timestamp-based version string
+package version
+
+import (
+	"path"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const versionFormat = "-v2006-01-02-150405.000"
```

Comment:
The exact number of significant figures here the `.000` depends on the backend. For b2 it is 3, if we implement this for AWS s3 it would probably be 9.

However 3 sig figs is probably enough and we shouldn't worry about it.